### PR TITLE
[subset] Limit the number of scripts and langsys' in feature collection.

### PR DESCRIPTION
**Pull request by garretrieger**

### CODE REVIEW

File: src/hb-ot-layout-common.hh [hunk 1]
```diff
@@ -51,6 +51,14 @@
 #define HB_CLOSURE_MAX_STAGES	32
 #endif
 
+#ifndef HB_MAX_SCRIPTS
+#define HB_MAX_SCRIPTS	1000
```
Comment (behdad) [inline src/hb-ot-layout-common.hh]: 200 here should be plenty.

File: src/hb-ot-layout.cc [hunk 2]
```diff
@@ -678,6 +679,10 @@ struct hb_collect_features_context_t
 		  !s.get_lang_sys_count ()))
       return true;
 
+    script_count++;
+    if (script_count > HB_MAX_SCRIPTS)
```
Comment (behdad) [inline src/hb-ot-layout.cc]: `if (script_count++` please.

File: src/hb-ot-layout.cc [hunk 3]
```diff
@@ -688,6 +693,10 @@ struct hb_collect_features_context_t
 		  !l.get_feature_count ()))
       return true;
 
+    langsys_count++;
+    if (langsys_count > HB_MAX_LANGSYS)
```
Comment (behdad) [inline src/hb-ot-layout.cc]: Ditto.

File: src/hb-ot-layout.cc [hunk 4]
```diff
@@ -710,6 +719,8 @@ struct hb_collect_features_context_t
   private:
   hb_set_t visited_script;
   hb_set_t visited_langsys;
+  unsigned int langsys_count;
+  unsigned int script_count;
```
Comment (behdad) [inline src/hb-ot-layout.cc]: Nit. Reorder the two so follows previous lines.

File: src/hb-ot-layout-common.hh (hunk 1 above)
Comment (behdad) [reply src/hb-ot-layout-common.hh]: Okay maybe 500 here.

Comment (behdad) [reply src/hb-ot-layout-common.hh]: And 2000 for language.

Comment (garretrieger) [reply src/hb-ot-layout-common.hh]: Done.

File: src/hb-ot-layout.cc (hunk 2 above)
Comment (garretrieger) [reply src/hb-ot-layout.cc]: Done.

File: src/hb-ot-layout.cc (hunk 4 above)
Comment (garretrieger) [reply src/hb-ot-layout.cc]: Done.

**Pull request merged**